### PR TITLE
Notarize app (mac)

### PR DESCRIPTION
**Pull request by steffenagger**
### CHANGES:
This PR adds `electron-notarize` & uses it to notarize the mac .app.
This requires a `hardenedRuntime` - configs updated accordingly (as far as I can determine).
A bug in `electron-builder` makes the zipped output useless with no easy workaround, so zipped releases for mac has been disabled (for now).

### TODOS:
- [x] Update changelog.
- [x] Update [realm@10.0.0 PR](https://github.com/realm/realm-studio/pull/1340) & rebase on it.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,12 +28,13 @@
       "!node_modules/realm/src${/*}",
       "!node_modules/realm/vendor${/*}"
     ],
+    "afterSign": "./scripts/afterSign",
     "mac": {
-      "hardenedRuntime": false,
+      "hardenedRuntime": true,
+      "gatekeeperAssess": false,
```
Comment (kraenhansen) [inline package.json:34]: Any reason this is `false`? Also - it looks to me as if this is `false` by default? https://www.electron.build/configuration/mac

File: scripts/afterSign.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+exports.default = async (context) => {
+  console.log("electron-builder.afterSign hook triggered.");
+
+  switch (context.electronPlatformName) {
+    case "darwin":
+      await notarizeMacApp(context);
+      break
+  }
+};
```
Comment (kraenhansen) [inline scripts/afterSign.js:11]:
```suggestion
};

```

File: scripts/mac/notarize.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  // Sanity check
+  if (!fs.existsSync(appPath)) {
+    throw new Error(`File not found: ${appPath}`);
+  }
+
+  await notarize(options);
+
+  console.log(`Done notarizing "${appBundleId}".`);
+};
```
Comment (kraenhansen) [inline scripts/mac/notarize.js:49]:
```suggestion
};

```

**Review by kraenhansen — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (steffenagger) [reply package.json:34]: Yes it's now `false` by default, but if true, `electron-builder` will test prematurely. (it could be omitted if preferred?)

**Pull request merged**